### ISSUE TITLE
[Reactivity Transform] `defineProps` destructure with object type defaults are not compiled correctly

### ISSUE THREAD

**Issue by CyanSalt** [vue-macros/vue-macros#279]
### Describe the bug

For code

```js
const { values = [] } = defineProps<{
  values?: number[],
}>()
```

the expectation is that it should be compiled to

```js
defineComponent({
  props: {
    values: { type: Array, required: false, default: () => [] }
    //                                               ^^
  },
})
```

but it is currently compiled to

```js
defineComponent({
  props: {
    values: { type: Array, required: false, default: [] }
  },
})
```

This can cause problems in some edge cases

![20230222162901_rec_](https://user-images.githubusercontent.com/5101076/220564681-0570da2f-31ee-4398-bb1e-50ef48a87bc2.gif)

---

Compiling object type defaults as functions is technically possible because `@vue/compiler-sfc` doesn't actually allow passing local variables to `withDefaults`. e.g.

<img width="924" alt="image" src="https://user-images.githubusercontent.com/5101076/220565780-c0438408-79d1-4d0a-a7b3-92cc0d9146a7.png">

Reactivity Transform should also consider literals only.

### Reproduction

https://codesandbox.io/p/github/CyanSalt/vue-macros-repro/with-defaults

### System Info

```Shell
Any
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sxzz** [comment]
It's by design.

https://github.com/sxzz/unplugin-vue-macros/blob/cb525c6bc2a7703323f21ffbce4037f8540143fa/packages/reactivity-transform/tests/fixtures/vue3/default-value.vue#L7

Further read: https://github.com/vuejs/core/issues/7150#issuecomment-1317143863

**Comment by CyanSalt** [comment]
I request to reopen this issue because Reactive Props Destructure in Vue 3.5 [has different behavior](https://play.vuejs.org/#eNp9kctOwzAQRX9l8CYglUYFVlVaBKgSsIAKkFjUXYRkkqY4tuVHqBTl35m49LFA3Xnm3rHP9bTsTuth45GNWWIzU2kHIpXlhDNnOQOLzuspl1WtlXHQgsECOiiMqiGisYhLLjMlba81qfBoYQKL0QCuluSbQI5FJXFulLZJyyX8mW7HIH39hWax5LKbnl8c7qltSXP0znn0iEIo+FRG5GcRWZJ4i0hAVDistUgdUgWQrEbTtg3DXZfEVIVuJbV30FzWKkdBoUinVDGJSXw0zwYUl54vqnK4tkrSbwRYzjJV60qgedWuIjzOxhCUXkuJ7uc59JzxONj1sxVm3//013bT9zibG7RoGuRsr7nUlOi28uz9BTd03otE7wW5T4hvaJXwPePWdu9lTthHvkD7FPZYyfLDzjYOpd2F6kF7Zxf8nNFuH05EP+BeD2/CHK2Rdb81zcOb) from this.

**Comment by sxzz** [comment]
Fixed via e27466c5c50f8593b376ec6f0d60a517a1848bbb
